### PR TITLE
Support for adding custom tika configuration

### PR DESCRIPTION
**Pull request by ahilmathew**
Added support for custom tika [configuration](https://tika.apache.org/1.9/configuring.html) to be used in the helm chart via the `tikaConfig` key in the `values.yaml`. 

This key is optional which when configured will create a `configMap` and then mounted as `tika-config.xml` into the running pod.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,6 +89,27 @@ while true; do kubectl --namespace tika-test port-forward $POD_NAME 9998:$CONTAI
 * Install it:
   - with Helm 3: `helm install tika . --set image.tag=latest-full`
 
+### Custom configuration for tika
+
+To use custom [configuration](https://tika.apache.org/1.9/configuring.html) values for apache tika, use the `tikaConfig` key in the `values.yaml`.
```
Comment (lewismc) [inline README.md:94]: The only suggestion I have is to point to more recent documentation.
https://tika.apache.org/2.9.1/configuring.html

**Review by lewismc — CHANGES_REQUESTED** [review]
This is a really nice patch @ahilmathew nice job! One minor suggestion.

File: README.md [hunk 2]
```diff
@@ -84,6 +89,27 @@ while true; do kubectl --namespace tika-test port-forward $POD_NAME 9998:$CONTAI
...
+tikaConfig: |
+  <?xml version="1.0" encoding="UTF-8"?>
+  <properties>
+    <parsers>
+      <!-- Default Parser for most things, except for 2 mime types -->
+      <parser class="org.apache.tika.parser.DefaultParser">
+        <mime-exclude>image/jpeg</mime-exclude>
+        <mime-exclude>application/pdf</mime-exclude>
+      </parser>
+      <!-- Use a different parser for PDF -->
```
Comment (nddipiazza) [inline README.md:106]: why do we want this empty parser here?

**Review by nddipiazza — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (lewismc) [reply README.md:106]: Good catch, we should just keep the default.

File: README.md (hunk 1 above)
Comment (ahilmathew) [reply README.md]: Thanks for the review. I've updated it.

File: README.md (hunk 2 above)
Comment (ahilmathew) [reply README.md]: Thanks for the review. I've updated it.

**Review by lewismc — APPROVED** [review]
Tested locally via k3d. LGTM.
Any thoughts @nddipiazza ?

**Comment by lewismc** [comment]
@nddipiazza any feedback? Would be great to get this merged if possible. Thanks

**Comment by lewismc** [comment]
@ahilmathew even if it is commented out by default, I think we should add the `tikaConfig` element to `values.yaml`.

**Comment by t-l-k** [comment]
@lewismc @nddipiazza I'm champing at the bit to to see this merged, xml configuration essential in Tika v2+

**Comment by lewismc** [comment]
@ahilmathew please sync with `main` branch and also include the default oncfiguration in `values.yaml`. Thank you

**Comment by ahilmathew** [comment]
@lewismc Thanks for the comments. I've updated `values.yaml` to include the configuration from https://tika.apache.org/2.9.1/configuring.html. Let me know if there is a specific configuration you would like to be included as default.

File: templates/configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.tikaConfig }}
```
Comment (lewismc) [inline templates/configmap.yaml:1]: @ahilmathew please add Apache License v2 file header. Thank you

**Review by lewismc — CHANGES_REQUESTED** [review]
Thanks @ahilmathew please see minor comment

File: templates/configmap.yaml (hunk 3 above)
Comment (ahilmathew) [reply templates/configmap.yaml:19]: @lewismc Done.

**Comment by ahilmathew** [comment]
@lewismc - Updated now.

**Review by lewismc — APPROVED** [review]
+1

**Comment by lewismc** [comment]
Thank you @ahilmathew really nice patch.

**Pull request merged**